api/client/trust.go

```diff
@@ -375,34 +375,52 @@ func (cli *DockerCli) trustedPush(repoInfo *registry.RepositoryInfo, tag string,
 
 	defer responseBody.Close()
 
-	targets := []target{}
+	// If it is a trusted push we would like to find the target entry which match the
+	// tag provided in the function and then do an AddTarget later.
+	target := &client.Target{}
+	// Count the times of calling for handleTarget,
+	// if it is called more that once, that should be considered an error in a trusted push.
+	cnt := 0
 	handleTarget := func(aux *json.RawMessage) {
+		cnt++
```

Comment:
Are you sure? The only place I see `handleTarget` being called is at line 410 so it seems multiple calls would have already happened. I think I misinterpreted your comment on line 396. I initially thought of it as "call it only once and refuse to call it more than that but don't error", which is not quite the behaviour you describe. In that case @HuKeping an int is the right type, but we could add a check immediately after this increment and skip running this function again if the int is already > 1.